File: core/src/main/java/com/graqr/threshr/model/redsky/product/SearchResponse.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record SearchResponse(
        Metadata metadata,
        @JsonProperty("facet_list")
        List<Facet> facets,
        @JsonProperty("visual_facet_list")
        List<Facet> visualFacets) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/product/SearchResponse.kt
package com.graqr.threshr.circleStore.model.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class SearchResponse(
    val metadata: Metadata,
    @param:JsonProperty("facet_list") val facets: List<Facet>,
    @param:JsonProperty("visual_facet_list") val visualFacets: List<Facet>
) {}
